### PR TITLE
Upgrade truffle, fix tests

### LINKED ISSUE #28: Tests & deployment broken
This is what we get for not running CI.. :smile:  (because #13)

I'm not sure when this happened exactly, but the test suite doesn't pass.  Like, more than is described in #13. There are some problems with `migrations/2_deploy_contracts.js` that will prevent it from running at all -- web3 is used, but never imported nor a web3 object created, and the web3 version pinned in package.json has a different API than the code used in there anyway (e.g. it seems `web3.fromAscii` should be used in place of `web3.utils.fromAscii`).

But after fixing that up or checking out c950c6ae681b7fdaa5cb5e7e12e8d387d3751799, the last commit before those changes were made, the test suite doesn't pass.  Even checking out v1.2.1, the tests don't pass.  I haven't checked back any further yet.

/cc @liam-fitzgerald @Fang-

### PR DESCRIPTION
**Pull request by pkova**
Fixes #28, #13, #15.

### CODE REVIEW

File: test/TestAzimuth.js [hunk 1]
```diff
@@ -182,30 +182,47 @@ contract('Azimuth', function([owner, user, user2, user3]) {
...
+                                    web3.utils.toHex(11),
+                                    web3.utils.toHex(2)), []);
+    let ks = await azimuth.getKeys(web3.utils.toHex(0));
+    crypt = ks.crypt;
+    auth = ks.auth;
+    suite = ks.suite;
+    revision = ks.revision;
     assert.equal(crypt,
-      '0xa000000000000000000000000000000000000000000000000000000000000000');
+      '0x0a00000000000000000000000000000000000000000000000000000000000000');
```
Comment (pkova) [inline test/TestAzimuth.js:214]: These right shifts are weird, happen in three places and no idea what causes this! Everything else is just boring breaking api changes.

**Comment by pkova** [comment]
Ping @jtobin.

File: test/TestAzimuth.js (hunk 1 above)
Comment (Fang-) [reply test/TestAzimuth.js:214]: Looks like it's right-padding on the byte rather than half-byte or string level now?

**Comment by Fang-** [comment]
Thanks for this, @pkova! In case you haven't already, probably a good idea to see how these dependency changes interact with Bridge's repo & testing/dev scripts.

**Pull request merged**